File: Sources/lambda/Services/VersionService.swift
import AWSLambdaEvents
import AWSLambdaRuntime
import Foundation
import Logging

struct VersionResponse: Codable {
    let version: String
    let deployedAt: String
    let deployedBy: String
    let gitSha: String
}

actor VersionService {
    func handleVersion() async throws -> APIGatewayV2Response {
        let response = VersionResponse(
            version: Environment.Deploy.version,
            deployedAt: Environment.Deploy.deployedAt,
            deployedBy: Environment.Deploy.deployedBy,
            gitSha: Environment.Deploy.gitSha
        )
        return try ResponseBuilder.createResponse(body: response, includeCacheControl: false)
    }
}

File: src/Lho.Lambda/Services/VersionService.cs
using Lho.Lambda.Models;
using Lho.Lambda.RuntimeConfiguration.Options;

namespace Lho.Lambda.Services;

public class VersionService(DeploymentOptions deployment)
{
  public VersionResponse GetVersion()
  {
    return new VersionResponse(
      Version: deployment.Version,
      DeployedAt: deployment.DeployedAt,
      DeployedBy: deployment.DeployedBy,
      GitSha: deployment.GitSha);
  }
}
